blog/03-11-2023-what-are-pfp-nfts-and-how-to-create-them/what-are-pfp-nfts-and-how-to-create-them.md

```diff
@@ -0,0 +1,353 @@
+
+### Coding for Smart Contract
+
+:::tip
+To learn more about setting up the environment to create PFP NFTs , check out the **[step-by-step guide to deploying your first full-stack dApp on Celo](https://docs.celo.org/blog/tutorials/step-by-step-guide-to-deploying-your-first-full-stack-dapp-on-celo#steps-to-set-up-the-truffle-project-and-its-configs)**, which provides detailed instructions on getting started.
+:::
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Strings.sol"; 
+
+contract FACEPFP is ERC721, Ownable {
+    using Counters for Counters.Counter;
+
+    Counters.Counter private _tokenIdCounter;
+
+    constructor() ERC721("FACEPFP", "PFP") {}
+
+    function _baseURI() internal pure override returns (string memory) {
+        return
+            "https://cloudflare-ipfs.com/ipfs/QmaczyZ79XE7MJ37TEAdngNXtBTxouxpitaCLCiTQdwkih/";
+    }
+// Change the CID here With Your Json Folder CID
+    function safeMint() public onlyOwner {
+        uint256 tokenId = _tokenIdCounter.current();
+        require(tokenId + 1 < totalSupply(), "Total supply reached");
+        _tokenIdCounter.increment();
+        _safeMint(msg.sender, tokenId);
+    }
+
+    function totalSupply() public pure returns (uint256) {
+        return 50; // set the total supply here
+    }
+
+    function tokenURI(
+        uint256 tokenId
+    ) public view override returns (string memory) {
+        if (!_exists(tokenId)) revert("URI query for nonexistent token");
+        return
+            string(abi.encodePacked(_baseURI(), Strings.toString(tokenId), ".json"));
+    }
+
+}
+```
+
+**The smart contract defines several functions that are used to create Mint NFTs, Here is the Explaination of those functions.**
+
+- The _baseURI() function is an internal function that returns the base URI for the token metadata. In this case, it returns a hardcoded URI that points to a JSON file stored on IPFS.
+
+- The safeMint() function is a public function that allows the owner of the contract to create new NFTs. Each NFT has a unique ID which is incremented by one each time a new NFT is created. The totalSupply() function returns the total number of NFTs that can be created, which in this case is 50.
+
+- The tokenURI() function returns the metadata URI for a given token ID. It concatenates the base URI, the token ID and the extension ".json" to form the complete URI. If the token ID does not exist, the function will revert with an error message.
+
+:::tip
+Use the [OpenZeppelin Wizard](https://wizard.openzeppelin.com/) to easily create, deploy, and interact with smart contracts using the OpenZeppelin library.
+:::
+

```

Comment:
Hey in such type of coding tutorials it would be better if we provide git repo of the src code, So if you have that make sure to add here